modules/reindex/src/main/java/org/elasticsearch/index/reindex/ReindexTask.java

```diff
@@ -326,5 +332,28 @@ public void onFailure(Exception e) {
                     });
             }
         }
+
+        private void setDone(ReindexTaskStateDoc state, ActionListener<ReindexTaskState> listener) {
+            // TODO: Maybe just normal acquire
+            semaphore.acquireUninterruptibly();
+            isDone = true;
+            long term = lastState.getPrimaryTerm();
+            long seqNo = lastState.getSeqNo();
+            reindexIndexClient.updateReindexTaskDoc(getPersistentTaskId(), state, term, seqNo, new ActionListener<>() {
+                @Override
+                public void onResponse(ReindexTaskState taskState) {
+                    lastState = taskState;
```

Comment:
I could be inclined to simply set `lastState = null` here? I think we do not want further updates to happen now?